.pre-commit-config.yaml

```diff
@@ -1,23 +1,64 @@
+---
 repos:
--   repo: git@github.com:Yelp/detect-secrets
+  - repo: git@github.com:Yelp/detect-secrets
     rev: v0.13.0
     hooks:
-    -   id: detect-secrets
-        args: ['--baseline', '.secrets.baseline']
+      - id: detect-secrets
+        args: [--baseline, .secrets.baseline]
 
--   repo: https://github.com/pre-commit/pre-commit-hooks
+  - repo: https://github.com/pre-commit/pre-commit-hooks
     rev: v2.3.0
     hooks:
-    -   id: check-json
-    -   id: check-toml
-    -   id: check-yaml
+      - id: check-json
+      - id: check-toml
+      - id: check-yaml
         exclude: .gitlab-ci.yml
-    -   id: end-of-file-fixer
-    -   id: fix-encoding-pragma
+      - id: end-of-file-fixer
+      - id: fix-encoding-pragma
         args: [--remove]
-    -   id: no-commit-to-branch
+      - id: no-commit-to-branch
         args: [--branch, develop, --branch, master, --pattern, release/.*]
-    -   id: pretty-format-json
+      - id: pretty-format-json
         args: [--autofix, --no-sort-keys]
-    -   id: trailing-whitespace
+      - id: trailing-whitespace
         args: [--markdown-linebreak-ext=md]
+
+  - repo: local
+    hooks:
+      - id: replacer
+        # More chars can be added
+        name: Replace unicode dashes with ascii dashes.
+        language: system
+        pass_filenames: false
+        entry: |
+          sh -c "
+          # echo -n '–' | hexdump -C # > e2 80 93
+          LC_ALL=C
+          for f in $(ls src/gdcdictionary/schemas/*.yaml); do
+            # echo 'Converting ' $f
+            sed -i '' 's/\xe2\x80\x93/-/g' $f
+          done
+          "
```

Comment:
1. This isn't going to work for everyone, such as Windows users. 
2. If we are going to enforce ascii, we could use [require-ascii](https://github.com/jumanjihouse/pre-commit-hooks#require-ascii). It's catch much more than just the hyphen. We'll need to exclude [these files](https://github.com/NCI-GDC/gdcdictionary/blob/4068e3d6dc287a4251f64670cb89aa2469407b5b/src/gdcdictionary/python/__init__.py#L34-L38), as described in [this comment](https://github.com/NCI-GDC/gdcdictionary/blob/4068e3d6dc287a4251f64670cb89aa2469407b5b/src/gdcdictionary/python/__init__.py#L66-L67).

Reply:
> 1. This isn't going to work for everyone, such as Windows users.

Good point. Should it be a deal breaker?

> 2. If we are going to enforce ascii, we could use [require-ascii](https://github.com/jumanjihouse/pre-commit-hooks#require-ascii). It's catch much more than just the hyphen. We'll need to exclude [these files](https://github.com/NCI-GDC/gdcdictionary/blob/4068e3d6dc287a4251f64670cb89aa2469407b5b/src/gdcdictionary/python/__init__.py#L34-L38), as described in [this comment](https://github.com/NCI-GDC/gdcdictionary/blob/4068e3d6dc287a4251f64670cb89aa2469407b5b/src/gdcdictionary/python/__init__.py#L66-L67).

The catch is that unicode chars are allowed. Depending on the version of the yaml file (which we do NOT explicitly define) and the version of the parser (somehow loose atm), one should either quote them or do whatever applies. 

When I first saw the failure and the traceback, I was inclined to remove the whole "ascii" enforcement and instead specify yaml versions and straighten up all the versions (yaml files and parser). 

I think we should pursue that end. It might take some bit effort but as long as we can diff the current json schema with the new one and make sure the match, we should be ok. IMHO.